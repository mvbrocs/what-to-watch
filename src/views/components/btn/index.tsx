import React, { ReactNode } from 'react';

type BtnProps = {
  children: ReactNode;
  onClick: () => void;
};

const Btn = ({ children, onClick }: BtnProps) => {
  return (
    <button className="btn movie-card__button" type="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Btn;
