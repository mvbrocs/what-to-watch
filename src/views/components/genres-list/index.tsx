import { MouseEventHandler } from 'react';

export type GenreClickHandler = (genre: string) => MouseEventHandler<HTMLAnchorElement>;
export { GenresList } from './genres-list';
