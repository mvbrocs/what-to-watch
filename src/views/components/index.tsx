export { default as GenresList } from './genres-list';
export { default as Header } from './header';
export { default as Logo } from './logo';
export { default as MovieCard } from './movie-card';
export { default as MovieList } from './movie-list';
export { default as Tabs } from './tabs';
export { default as TabPane } from './tabs/tab-pane';
export { default as UserBlock } from './user-block';
export { default as VideoPlayer } from './video-player';
export { default as VideoPlayerFullscreen } from './video-player-fullscreen';
export { default as Btn } from './btn';
