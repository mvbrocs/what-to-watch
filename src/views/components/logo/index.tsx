import React from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

import { Path } from 'src/routes';

type LogoProps = {
  light?: boolean;
};

const Logo = ({ light }: LogoProps) => (
  <div className="logo">
    <Link to={Path.HOME} className={classNames('logo__link', { 'logo__link--light': light })}>
      <span className="logo__letter logo__letter--1">W</span>
      <span className="logo__letter logo__letter--2">T</span>
      <span className="logo__letter logo__letter--3">W</span>
    </Link>
  </div>
);

export default Logo;
