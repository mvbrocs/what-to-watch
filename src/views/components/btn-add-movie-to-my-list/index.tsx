import React from 'react';

import { Btn } from 'src/views/components';
import { AddIcon, InListIcon } from 'src/views/icons';
import { IFilm } from 'src/api/films';
import { useBtnAddMovieToMyList } from './use-btn-add-movie-to-my-list';

type BtnAddMovieToMyListProps = {
  movie: IFilm | null;
};

const BtnAddMovieToMyList = ({ movie }: BtnAddMovieToMyListProps) => {
  const { movieInList, clickHandler } = useBtnAddMovieToMyList(movie);
  const btnAddIcon = movieInList ? <InListIcon /> : <AddIcon />;

  return (
    <Btn onClick={clickHandler}>
      {btnAddIcon}
      <span>My list</span>
    </Btn>
  );
};

export default BtnAddMovieToMyList;
