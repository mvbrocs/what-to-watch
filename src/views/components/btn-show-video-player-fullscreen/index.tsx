import React from 'react';

import { Btn } from 'src/views/components/index';
import { PlaySIcon } from 'src/views/icons';
import { useBtnShowVideoPlayerFullscreen } from './use-btn-show-video-player-fullscreen';
import { IFilm } from 'src/api/films';

type BtnShowVideoPlayerFullscreenProps = {
  movie: IFilm | null;
};

const BtnShowVideoPlayerFullscreen = ({ movie }: BtnShowVideoPlayerFullscreenProps) => {
  const { clickHandler } = useBtnShowVideoPlayerFullscreen(movie);

  return (
    <Btn onClick={clickHandler}>
      <PlaySIcon />
      <span>Play</span>
    </Btn>
  );
};

export default BtnShowVideoPlayerFullscreen;
