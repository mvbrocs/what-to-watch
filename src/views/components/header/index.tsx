import React, { ReactNode } from 'react';
import classNames from 'classnames';

import { Logo } from 'src/views/components';

type HeaderProps = {
  className?: string;
  children?: ReactNode;
};

const Header = ({ className, children }: HeaderProps) => (
  <header className={classNames('page-header', className)}>
    <Logo />

    {children}
  </header>
);

export default Header;
