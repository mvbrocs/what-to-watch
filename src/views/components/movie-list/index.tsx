import React from 'react';

import { MovieCard } from 'views/components/movie-card';
import { IFilm } from 'api/films';

type MovieListProps = {
  movies: IFilm[];
};

export const MovieList = ({ movies }: MovieListProps) => (
  <div className="catalog__movies-list">
    {movies.map((movie) => (
      <MovieCard {...movie} key={movie.id} />
    ))}
  </div>
);
