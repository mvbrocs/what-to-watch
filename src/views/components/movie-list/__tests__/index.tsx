import React from 'react';
import { shallow } from 'enzyme';

import { MovieList } from 'src/views/components';
import { mockFilms } from 'src/mocks';

describe('MovieList', () => {
  const movieList = shallow(<MovieList movies={mockFilms} />);

  it('renders properly', () => {
    expect(movieList).toMatchSnapshot();
  });
});
