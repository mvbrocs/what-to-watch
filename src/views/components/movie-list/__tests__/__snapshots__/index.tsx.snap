// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders properly 1`] = `
<div
  className="catalog__movies-list"
>
  <MovieCard
    id={1}
    key="1"
    name="Orlando"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/orlando.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={2}
    key="2"
    name="Beach"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/beach.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={3}
    key="3"
    name="Midnight Special"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/midnight-special.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={4}
    key="4"
    name="Once Upon a Time in America"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/Once_Upon_a_Time_in_America.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={5}
    key="5"
    name="Seven Years in Tibet"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/seven-years-in-tibet.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={6}
    key="6"
    name="Gangs of new york"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={7}
    key="7"
    name="Macbeth"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/macbeth.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={8}
    key="8"
    name="Moonrise Kingdom"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={9}
    key="9"
    name="A Star Is Born"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/A_Star_Is_Born.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={10}
    key="10"
    name="Bohemian Rhapsody"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/bohemian_rhapsody.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={11}
    key="11"
    name="Fantastic Beasts: The Crimes of Grindelwald"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={12}
    key="12"
    name="What We Do in the Shadows"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={13}
    key="13"
    name="Pulp Fiction"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/pulp-fiction.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={14}
    key="14"
    name="Matrix"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/matrix.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={15}
    key="15"
    name="Aviator"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/aviator.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={16}
    key="16"
    name="Legend"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/legend.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={17}
    key="17"
    name="The Revenant"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/revenant.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={18}
    key="18"
    name="War of the Worlds"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={19}
    key="19"
    name="No Country for Old Men"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/no-country-for-old-men.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={20}
    key="20"
    name="We need to talk about Kevin"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/we-need-to-talk-about-kevin.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={21}
    key="21"
    name="Dardjeeling Limited"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/dardjeeling_limited.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={22}
    key="22"
    name="Johnny English"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/johnny-english.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={23}
    key="23"
    name="Snatch"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/snatch.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
  <MovieCard
    id={24}
    key="24"
    name="Bronson"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/bronson.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
  <MovieCard
    id={25}
    key="25"
    name="Shutter Island"
    previewImage="https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/shutter-island.jpg"
    previewVideoLink="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
</div>
`;
